### PR TITLE
docs: Initial k8s-executor notes

### LINKED ISSUE #236: document how to use `KubernetesExecutor`
Currently we natively support the `KubernetesExecutor` and `CeleryKubernetesExecutor` executors, but there is nothing to reflect this in the main `README.md`.

We should add a new section called something like: `How to use KubernetesExecutor?`, and outline the values which are specific to that mode.

### PR DESCRIPTION
**Pull request by whatnick**
Signed-off-by: Tisham Dhar <whatnickd@gmail.com>

**What issues does your PR fix?**

- partially fixes #236

**What does your PR do?**

Adds documentation for `KubernetesExecutor`.

## Checklist
- [x] Commits are [signed](https://github.com/airflow-helm/charts/tree/main/CONTRIBUTING.md#sign-your-work)
- [x] Commits are [squashed](https://github.com/airflow-helm/charts/tree/main/CONTRIBUTING.md#squash-commits) (if appropriate)
- [ ] Chart [version bumped](https://github.com/airflow-helm/charts/tree/main/CONTRIBUTING.md#versioning)
- [x] Chart [documentation updated](https://github.com/airflow-helm/charts/tree/main/CONTRIBUTING.md#documentation)
- [ ] Passes [linting](https://github.com/airflow-helm/charts/tree/main/CONTRIBUTING.md#linting)

### CODE REVIEW

File: charts/airflow/README.md [hunk 1]
```diff
@@ -758,6 +758,15 @@ serviceAccount:
 <hr>
 </details>
 
+### How to use `KubernetesExecutor` and `CeleryKubernetesExecutor`?
```
Comment (thesuperzapper) [inline charts/airflow/README.md:761]: I was thinking we should give an example of a values.yaml which will enable `KubernetesExecutor`, (and probably a separate one for `CeleryKubernetesExecutor`, as that needs all the normal Celery Stuff).


For example,
```yaml
airflow:
  executor: KubernetesExecutor

  kubernetesPodTemplate:
    ...

workers:
  enabled: false

redis:
  enabled: false
```

Comment (whatnick) [reply charts/airflow/README.md:761]: Working on that next, I will squash this PR down to one commit when ready or would you like to do that on merge ?

Comment (whatnick) [reply charts/airflow/README.md]: Made suggested updates. Would prefer to make the sample values.yaml a separate PR if that is acceptable.

File: charts/airflow/README.md [hunk 2]
```diff
@@ -803,6 +803,57 @@ serviceAccount:
 <hr>
 </details>
 
+### How to use `KubernetesExecutor`?
```
Comment (thesuperzapper) [inline charts/airflow/README.md:806]: You forgot to wrap in <details> tags.

**Comment by thesuperzapper** [comment]
@whatnick make sure your commits are "semantic" e.g. they must be named `docs: xxxx`

**Comment by thesuperzapper** [comment]
I also think we should be a lot more descriptive about the differences between Celery and Kubernetes executor modes, and the impact it will have for users (assume people are new-ish to airflow itself).

**Comment by whatnick** [comment]
I am concerned about how much of the main aiflow docs to replicate here. I have made some notes around the importance of the `pod_template_file`, please let me know what you think. I foresee a few more commits to get this to a mutually agreeable place. I will keep pushing to the branch and squash into a docs commit message when ready.

**Pull request closed without merging**